tests/TestCase/BEditaClientTest.php

```diff
@@ -837,6 +869,259 @@ public function testSchema()
         static::assertNotEmpty($response);
     }
 
+    /**
+     * Data provider for `testDelete`
+     */
+    public function responseBodyProvider()
+    {
+        return [
+            'get users' => [
+                [
+                    'method' => 'GET',
+                    'path' => '/users',
+                    'query' => null,
+                    'headers' => null,
+                    'body' => null,
+                ],
+                [
+                    'code' => 200,
+                    'message' => 'OK',
+                    'fields' => ['data', 'links', 'meta'],
+                ],
+            ],
+            'get unexisting user' => [
+                [
+                    'method' => 'GET',
+                    'path' => '/users/9999999',
+                    'query' => null,
+                    'headers' => null,
+                    'body' => null,
+                ],
+                new BEditaClientException('[404] Not Found', 404),
+            ],
+        ];
+    }
+
+    /**
+     * Test `getResponseBody`.
+     *
+     * @return void
+     *
+     * @covers ::getResponseBody()
+     * @dataProvider responseBodyProvider()
+     */
+    public function testGetResponseBody($input, $expected)
+    {
+        $response = $this->myclient->authenticate($this->adminUser, $this->adminPassword);
+        $this->myclient->setupTokens($response['meta']);
+
+        $exception = ($expected instanceof \Exception);
+        if ($exception) {
+            static::expectException(get_class($expected));
+            static::expectExceptionCode($expected->getCode());
+        }
+        $this->myclient->sendRequestRetry($input['method'], $input['path']);
+        $response = $this->myclient->getResponseBody();
+        if (!$exception) {
```

Comment:
check not necessary, if exception is thrown next statements will not be performed anyway